core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java

```diff
@@ -51,40 +57,21 @@ public class GrpcConnectionPool {
   public static final GrpcConnectionPool INSTANCE = new GrpcConnectionPool();
 
   /** gRPC Managed channels/connections. */
-  private ConcurrentMap<GrpcConnectionKey, CountingReference<ManagedChannel>> mChannels;
+  private final ConcurrentMap<GrpcConnectionKey, CountingReference<ManagedChannel>> mChannels
+      = new ConcurrentHashMap<>();
 
   /** Event loops. */
-  private ConcurrentMap<GrpcNetworkGroup, CountingReference<EventLoopGroup>> mEventLoops;
+  private final ConcurrentMap<GrpcNetworkGroup, CountingReference<EventLoopGroup>> mEventLoops
+      = new ConcurrentHashMap<>();
 
   /** Used to assign order within a network group. */
-  private ConcurrentMap<GrpcNetworkGroup, AtomicLong> mNetworkGroupCounters;
+  private final Map<GrpcNetworkGroup, AtomicLong> mNetworkGroupCounters
+      = Arrays.stream(GrpcNetworkGroup.values())
+      .collect(toImmutableMap(identity(), group -> new AtomicLong()));
 
   /** Used for obtaining SSL contexts for transport layer. */
-  private SslContextProvider mSslContextProvider;
-
-  /**
-   * Creates a new {@link GrpcConnectionPool}.
-   */
-  public GrpcConnectionPool() {
-    mChannels = new ConcurrentHashMap<>();
-    mEventLoops = new ConcurrentHashMap<>();
-    // Initialize counters for known network-groups.
-    mNetworkGroupCounters = new ConcurrentHashMap<>();
-    for (GrpcNetworkGroup group : GrpcNetworkGroup.values()) {
-      mNetworkGroupCounters.put(group, new AtomicLong());
-    }
-  }
-
-  /**
-   * @param conf Alluxio configuration
-   * @return Ssl context provider instance
-   */
-  private synchronized SslContextProvider getSslContextProvider(AlluxioConfiguration conf) {
-    if (mSslContextProvider == null) {
-      mSslContextProvider = SslContextProvider.Factory.create(conf);
-    }
-    return mSslContextProvider;
-  }
+  private final SslContextProvider mSslContextProvider =
```

Comment:
Do we always want to initialize the SslContextProvider?
Looks like SslContextProvider is not always needed.
Will it be expensive in some cases (for some SSL context other than the default one which does nothing)

Reply:
Double checked, it's not expensive, just storing the conf